fontforge/sfd.c

```diff
@@ -3696,6 +3726,51 @@ static ImageList *SFDGetImagePNG(FILE *sfd) {
 }
 #endif
 
+static ImageList *SFDGetImageReference(FILE *sfd) {
+    ImageList *img = calloc(1,sizeof(ImageList));
+    int ch;
+
+    getreal(sfd,&img->xoff);
+    getreal(sfd,&img->yoff);
+    getreal(sfd,&img->xscale);
+    getreal(sfd,&img->yscale);
+
+    if (ch = nlgetc(sfd) != '\n') { ungetc(ch, sfd); return NULL; }
+
+    char* hex = SFDReadUTF7Str(sfd);
```

Comment:
Check this function please, I think you're leaking multiple dynamically allocated objects.